components/dashboard/src/workspaces/ConnectToSSHModal.tsx

```diff
@@ -55,40 +57,80 @@ interface SSHProps {
 }
 
 function SSHView(props: SSHProps) {
-    const sshCommand = `ssh '${props.workspaceId}#${props.ownerToken}@${props.ideUrl.replace(
-        props.workspaceId,
-        props.workspaceId + ".ssh",
-    )}'`;
+    const [hasSSHKey, setHasSSHKey] = useState(true);
+    const [selectSSHKey, setSelectSSHKey] = useState(true);
+
+    useEffect(() => {
+        getGitpodService()
+            .server.hasSSHPublicKey()
+            .then((d) => {
+                setHasSSHKey(d);
+            })
+            .catch(console.error);
+    }, []);
+
+    const host = props.ideUrl.replace(props.workspaceId, props.workspaceId + ".ssh");
+    const sshAccessTokenCommand = `ssh '${props.workspaceId}#${props.ownerToken}@${host}'`;
+    const sshKeyCommand = `ssh '${props.workspaceId}@${host}'`;
+
     return (
-        <div className="border-t border-b border-gray-200 dark:border-gray-800 mt-2 -mx-6 px-6 py-6">
-            <div className="mt-1 mb-4">
-                <AlertBox>
-                    <p className="text-red-500 whitespace-normal text-base">
+        <>
+            <div className="flex flex-row">
+                <TabMenuItem
+                    key="ssh_key"
+                    name="SSH Key"
+                    selected={selectSSHKey}
+                    onClick={() => {
+                        setSelectSSHKey(true);
+                    }}
+                />
+                <TabMenuItem
+                    key="access_token"
+                    name="Access Token"
+                    selected={!selectSSHKey}
+                    onClick={() => {
+                        setSelectSSHKey(false);
+                    }}
+                />
```

Comment:
**question**: Do we want to keep the access token option around? I thought this caused more confusion than helped. Maybe it's worth removing this altogether and instead nudge users to add an SSH key when they try to connect via SSH and there's no SSH key in their account. Cc @akosyakov @loujaybee

Reply:
I think it is alright to have, but ssh key should be always default.

Reply:
> I think it is alright to have, but ssh key should be always default.

Fair point, let's keep it there then. 🤝 

If there's no reason to keep access tokens around, I'd vote for removing this option to limit the amount of options presented to users, as it can cause more confusion than value.

In case we keep this, does it make sense to drop the command printed when there are no keys in place?

<img width="571" alt="Screenshot 2022-06-30 at 12 38 19 AM" src="https://user-images.githubusercontent.com/120486/176549624-ec7a15bb-0f4f-429b-9a9d-cf7d948b6738.png">

Reply:
> In case we keep this, does it make sense to drop the command printed when there are no keys in place?

answered [here ](https://github.com/gitpod-io/gitpod/pull/10573#discussion_r909161251) right?